docs/contributor/draft/rate-limit-design.md

```diff
@@ -0,0 +1,80 @@
+# Rate Limit Proposal
+
+## Spec 
+
+| field                                            | type        | description                                                     | required                      |
+|--------------------------------------------------|-------------|-----------------------------------------------------------------|-------------------------------|
+| wokloadSelector                                  | object      | when skipped, it's applied for the whole namespace workloads, if empty and in istio-system ns, applies to all workloads in all namespaces | no                            |
+| workloadSelector.labels                          | object      | allows to select given workload by a label selector                   | no                            |
+| local                                   | object      | allows to describe local rate limit properties                  | yes                           |
+| local.bucket                            | object      | allows to set rate limit capacity through bucket concept        | yes                           |
+| local.bucket.max_tokens                 | int         | max tokens that can be gathered for the rate limit capacity     | yes                           |
+| local.bucket.tokens_per_fill            | int         | number of tokens added per fill cycle                           | yes                           |
+| local.bucket.fill_interval              | time        | interval on which tokens are filled                             | yes                           |
+
+## Usage example
+
+```yaml
+apiVersion: operator.kyma-project.io/v1alpha1
+kind: RateLimit
+metadata:
+  name: httpbin-local-rate-limit
+  namespace: default
+spec:
+  context: "gateway" | "sidecar" # translates to: PatchContext: GATEWAY | SIDECAR_INBOUND, should be on this level to not to mix sidecar level rate limit with gateway level rate limit. In such case, 2 separate resources should be created.
```

Comment:
In case gateway context is set, do we do some magic to apply it automatically to `istio-system` and `ingressgateway` or do we want to be expressive? I think having less "magic" for a starter might be better. We might want to mention this in the decision.

Reply:
We decided to not do any magic, but to use workload selector value with context to do the validation around it